src/func_surf_energy_bal.c

```diff
@@ -687,15 +690,65 @@ double func_surf_energy_bal(double Ts, va_list ap)
 			    TMean, Tair, wind[UnderStory], roughness[UnderStory]);
   else
     Ra_used[0] = HUGE_RESIST;
-  
+
+  /*************************************************
+    Compute aerodynamic resistance for the case of exposed soil between
+    plants (or gaps in canopy).  Assume plants and exposed soil are well-
+    mixed, i.e., exposed soil is neither as disconnected from the
+    atmosphere as soil under veg or canopy, nor as exposed as a large
+    area of exposed soil.  Rather, it is subject to some wind attenuation
+    from the surrounding plants.  Thus, compute as the area-weighted
+    average of "pure" understory and "pure" exposed conditions.
+
+    NOTE: can't average the resistances; must convert to conductances,
+    average the conductances, and then convert the average conductance
+    to a resistance.
+  *************************************************/
+  if (veg_var->vegcover < 1) {
+    /** If Ra_used (which pertains to under vegetation at this point)
+        is non-zero, use it to compute area-weighted average **/
+    if (Ra_used[0] > 0) {
+      /** aerodynamic conductance under vegetation **/
+      ga_veg = 1/Ra_used[0];
+      /** compute aerodynamic resistance over exposed soil (Ra_bare) **/
+      tmp_wind[0] = wind[0];
+      tmp_wind[1] = -999.; // unused
+      tmp_wind[2] = -999.; // unused
+      tmp_height = soil_con->rough/0.123; // eqn from calc_veg_roughness()
```

Comment:
OK, I'm going to replace the -999s with ERROR (which is defined as -999 in vicNl_def.h), both here and in full_energy(), which is where I copied these lines from.

I'm defining a few other constants too:
RATIO_DH_HEIGHT_VEG 0.67 - used in calc_veg_displacement() and calc_veg_height()
RATIO_RL_HEIGHT_VEG 0.123 - used both here and in calc_veg_roughness()
WINDH_SOIL 10.0